lib/base/logger.cpp

```diff
@@ -243,17 +244,35 @@ void Logger::UpdateMinLogSeverity()
 	m_MinLogSeverity.store(result);
 }
 
-Log::Log(LogSeverity severity, String facility, const String& message)
-	: Log(severity, std::move(facility))
+Log::Log(LogSeverity severity, String facility, const String& message, const std::vector<ConfigObject::Ptr>& involved)
+	: Log(severity, std::move(facility), involved)
 {
 	if (!m_IsNoOp) {
 		m_Buffer << message;
 	}
 }
 
-Log::Log(LogSeverity severity, String facility)
-	: m_Severity(severity), m_Facility(std::move(facility)), m_IsNoOp(severity < Logger::GetMinLogSeverity())
-{ }
+Log::Log(LogSeverity severity, String facility, const std::vector<ConfigObject::Ptr>& involved)
+	: m_Severity(severity), m_Facility(std::move(facility))
+{
+	std::set<ConfigObject::Ptr> objects;
+
+	for (auto& obj : involved) {
+		objects.merge(DependencyGraph::GetAllParents(obj));
+	}
+
+	objects.insert(involved.begin(), involved.end());
```

Comment:
But it would iterate an existing small set, not a computed one like this.

What do you think?